dspace-server-webapp/src/main/java/org/dspace/app/rest/DuplicateDetectionRestController.java

```diff
@@ -0,0 +1,134 @@
+import org.dspace.content.service.DuplicateDetectionService;
+import org.dspace.content.service.ItemService;
+import org.dspace.content.virtual.PotentialDuplicate;
+import org.dspace.core.Context;
+import org.dspace.discovery.SearchServiceException;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.data.web.PagedResourcesAssembler;
+import org.springframework.hateoas.Link;
+import org.springframework.hateoas.PagedModel;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * The controller for the api/duplicates endpoint, which handles requests for finding
+ * potential duplicates
+ *
+ * @author Kim Shepherd
+ */
+@RestController
+@RequestMapping("/api/" + PotentialDuplicateRest.CATEGORY)
+public class DuplicateDetectionRestController implements InitializingBean {
+
+    private static final Logger log = LogManager.getLogger(DuplicateDetectionRestController.class);
+
+    @Autowired
+    protected Utils utils;
+    @Autowired
+    private DiscoverableEndpointsService discoverableEndpointsService;
+    @Autowired
+    private ItemService itemService;
+    @Autowired
+    private DuplicateDetectionService duplicateDetectionService;
+    @Autowired
+    private ConverterService converter;
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        discoverableEndpointsService
+            .register(this, Arrays.asList(Link.of("/api/" + PotentialDuplicateRest.CATEGORY,
+                    PotentialDuplicateRest.CATEGORY)));
+    }
+
+    /**
+     * Return a paged list of potential duplicate matches for the given item ID. This may be an item wrapped in
+     * an in-progress item wrapper like workspace or workflow, as long as the current user has READ access to this item.
+     * Results from the service search method will only contain matches that lead to items which are readable by
+     * the current user.
+     *
+     * @param uuid The item UUID to search
+     * @param page Pagination options
+     * @param assembler The paged resources assembler to construct the paged model
+     * @return Paged list of potential duplicates
+     * @throws Exception
+     */
+    @RequestMapping(method = RequestMethod.GET, value = "/search")
+    @PreAuthorize("hasPermission(#uuid, 'ITEM', 'READ')")
+    public PagedModel<PotentialDuplicateResource> searchPotentialDuplicates(
+            @RequestParam(name = "uuid", required = true) UUID uuid, Pageable page, PagedResourcesAssembler assembler) {
```

Comment:
If I'm not mistaken, this method is **only called** during submission/workflow.  That makes me suspect we might want to rename this endpoint to be
```
/api/submission/duplicates/search?uuid=
```

Currently, it appears to be named:
```
/api/duplicates/search?uuid=
```

While I'm not against that name, it implies this should be used in **other features** as it's a top-level endpoint right under `/api`...instead of grouping it with other submission features under `/api/submission` (like `/api/submission/vocabularies` and similar)

Reply:
Although right now the only display of duplicates is in submission / workflow, I know of at least one group that plans to generate their own reports of archived items with duplicate titles, via REST calls, which is I wouldn't want to make it seem like this is a submission-only feature. (identitying duplicates can be a useful clean-up exercise as well as a warning during submission). Does that make sense?
I will change the path if required, just explaining my reasoning.

Reply:
@kshepherd : Makes sense.  I think the only reason I bring this up, is that most of our endpoints look like this:
```
/api/[category]/[endpoint-name]/search
```
But, this one looks like:
```
/api/[endpoint-name]/search
```

So, I was trying to figure out an existing "category" that it could go into.   It seemed odd to me that this endpoint is a category all to itself.  There's also the "tools" category (which is used by a few endpoints).   The "submission" one just jumped out as related since that's where it's currently used.

This isn't the highest priority thing, to be honest...I was merely looking for consistency with other REST endpoints.

Reply:
i have moved it to /api/submission/duplicates/search and updated names / angular paths / docs / rest contract accordingly